### ISSUE TITLE
Thread the downloads and show download progress

### ISSUE THREAD

**Issue by josefnpat** [josefnpat/vapor#1]
Use threads to download and update the main thread on the progress of the download. That way the main thread won't be blocked by the download.

**Comment by jessefromearth** [comment]
How would the download be represented graphically?

**Comment by josefnpat** [comment]
I was thinking the icons could rotate between 
https://github.com/josefnpat/vapor/blob/master/assets/icons/network-idle.png
https://github.com/josefnpat/vapor/blob/master/assets/icons/network-receive.png
https://github.com/josefnpat/vapor/blob/master/assets/icons/network-transmit-receive.png
https://github.com/josefnpat/vapor/blob/master/assets/icons/network-transmit.png

**Comment by qaisjp** [comment]
I was thinking more of a rotating icon (the one you get on iOS and mobiles these days) in place of the download/play button. and then a green progress bar underneath text but over the grey row showing the download progress. Hovering on the row will show "xx/unit / xx/unit" or "52%" (xx/unit can be 250kb/30mb depending on the amount or just stick to kb's)

**Comment by josefnpat** [comment]
@qaisjp at the moment, I'd rather use a universal set of icons, unless we can get an artist to remake all of them. Right now, we're using this icon set;

http://art.gnome.org/themes/icon/1100

See any you think would work better?

**Comment by qaisjp** [comment]
How about FamFamFam www.famfamfam.com/lab/icons/
Just a wild suggestion.

**Comment by josefnpat** [comment]
@qaisjp #9

**Comment by josefnpat** [comment]
1c20dfc

**Comment by qaisjp** [comment]
What about my progress bar/hover text?

**Comment by josefnpat** [comment]
Feel free to fork and contribute!

Might make sense to make a new ticket for this, as the purpose of this issue was to thread the downloads, and provide some indication that the game was being downloaded.
